feature: Add list like aggregation support for metadata

frascuchon opened the pull request:
<!-- Thanks for your contribution! As part of our Community Growers initiative 🌱, we're donating Justdiggit bunds in your name to reforest sub-Saharan Africa. To claim your Community Growers certificate, please contact David Berenstein in our Slack community or fill in this form https://tally.so/r/n9XrxK once your PR has been merged. -->

# Description

This PR adds list support for term metadata values.
 
Closes #4359

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor (change restructuring the codebase without changing functionality)
- [X] Improvement (change adding some improvement to an existing functionality)

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

Tested locally with this code snippet:
```python
dataset = rg.FeedbackDataset(
     fields=[rg.TextField(name="text"), rg.TextField(name="optional", required=False)],
     questions=[rg.TextQuestion(name="question")],
     metadata_properties=[
         rg.TermsMetadataProperty(name="terms-metadata", values=["a", "b", "c"]),
         rg.IntegerMetadataProperty(name="integer-metadata"),
         rg.FloatMetadataProperty(name="float-metadata", min=0.0, max=10.0),
     ],
 )

ds = dataset.push_to_argilla("ds", workspace="argilla")

records = [
   rg.FeedbackRecord(fields={"text": "Hello world!"}, metadata={"terms-metadata": "a"}),
   rg.FeedbackRecord(fields={"text": "Hello world!"}, metadata={"terms-metadata": ["b", "a"]}),
]

ds.add_records(records)
```


**Checklist**

- [ ] I added relevant documentation
- [X] I followed the style guidelines of this project
- [X] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [X] I have added relevant notes to the `CHANGELOG.md` file (See https://keepachangelog.com/)

frascuchon committed:
feat(client): changes to support list as valid terms metadata value

frascuchon committed:
feat(server): list support for terms metadata values

frascuchon committed:
tests: add new test

frascuchon committed:
chore: update Changelog

jfcalvo reviewed (approved):
@frascuchon do we need to change searching someway to support this?

davidberenstein1957 committed:
Merge branch 'develop' into 4359-feature-add-list-like-aggregation-support-for-metadata

kursathalat committed:
docs: Update documentation on metadata allowing for aggregate values (#4439)

<!-- Thanks for your contribution! As part of our Community Growers
initiative 🌱, we're donating Justdiggit bunds in your name to reforest
sub-Saharan Africa. To claim your Community Growers certificate, please
contact David Berenstein in our Slack community or fill in this form
https://tally.so/r/n9XrxK once your PR has been merged. -->

# Description

This PR adds info in the docs regarding the recent change on metadata
where aggregate metadata values can now be added to each record in the
form of a list.

Closes #4416 

**Type of change**

(Remember to title the PR according to the type of change)

- [ ] Documentation update

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes.)

- [ ] `sphinx-autobuild` (read [Developer
Documentation](https://docs.argilla.io/en/latest/community/developer_docs.html#building-the-documentation)
for more details)

**Checklist**

- [ ] I added relevant documentation
- [ ] I followed the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK)
(see text above)
- [ ] I have added relevant notes to the `CHANGELOG.md` file (See
https://keepachangelog.com/)

---------

Co-authored-by: David Berenstein <david.m.berenstein@gmail.com>
